src/onnx/parse_qlinearpooling.cpp

```diff
@@ -0,0 +1,120 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <migraphx/onnx/op_parser.hpp>
+#include <migraphx/onnx/pooling.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/op/pooling.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/onnx/checks.hpp>
+#include <migraphx/onnx/broadcast_qdq.hpp>
+#include <migraphx/instruction.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+/*
+ *********************************************************************************
+ *  Reference: see QLinearAveragePool and QLinearGlobalAveragePool in            *
+ *  github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md          *
+ *********************************************************************************
+ */
+
+struct parse_qlinearpooling : op_parser<parse_qlinearpooling>
+{
+    std::vector<op_desc> operators() const
+    {
+        return {{"QLinearGlobalAveragePool", "average"}, {"QLinearAveragePool", "average"}};
+    }
+
+    void check_inputs(const op_desc& opd, const std::vector<instruction_ref>& args) const
+    {
+        const auto& in_x     = args[0];
+        const auto onnx_name = opd.onnx_name;
+
+        if(in_x->get_shape().ndim() <= 2)
+            MIGRAPHX_THROW(onnx_name + ": input dimensions too small");
+
+        auto type_x = in_x->get_shape().type();
+        if(type_x != migraphx::shape::int8_type and type_x != migraphx::shape::uint8_type)
+            MIGRAPHX_THROW(onnx_name + ": unsupported input type");
+
+        // zero_pt_x and zero_pt_y is mandatory for QLinearAveragePool according to the spec
+        if(starts_with(onnx_name, "QLinearGlobal"))
+        {
+            if(args.size() < 5)
+                MIGRAPHX_THROW(onnx_name + ": missing inputs");
+
+            const auto& zero_pt_x = args[2];
+            const auto& zero_pt_y = args[4];
+
+            if(type_x != zero_pt_x->get_shape().type())
+                MIGRAPHX_THROW(onnx_name + ": mismatched type: input zero point");
+
+            if(type_x != zero_pt_y->get_shape().type())
+                MIGRAPHX_THROW(onnx_name + ": mismatched type: output zero point");
+        }
+    }
+
+    instruction_ref parse(const op_desc& opd,
+                          const onnx_parser& parser,
+                          const onnx_parser::node_info& info,
+                          const std::vector<instruction_ref>& args) const
+    {
+        if(contains(info.attributes, "channel_last"))
+        {
+            int channels_last =
+                parser.parse_value(info.attributes.at("channels_last")).template at<int>();
+            if(channels_last != 0)
+                MIGRAPHX_THROW(opd.onnx_name + ": channels_last (N x D1..Dn x C) is not supported");
+        }
+
+        check_inputs(opd, args);
+
+        // Input: X
+
+        const auto& in_x      = args[0];
+        const auto& scale_x   = args[1];
+        const auto& zero_pt_x = args[2];
```

Comment:
You check zero_pt_x only for qlinearglobalavg, and yet use it for qlinearavg also..

Reply:
^^ Checking  is in check_inputs().

Reply:
Added the checking for both operators